src/core/lib/channel/connected_channel.cc

```diff
@@ -548,7 +548,7 @@ ArenaPromise<ServerMetadataHandle> MakeClientCallPromise(
         }
         if (!done_recv_messages) {
           auto p = recv_messages();
-          if (auto* r = p.value_if_ready()) {
+          if ([[maybe_unused]] auto* r = p.value_if_ready()) {
```

Comment:
`[[maybe_unused]]` is a C++17 attribute. I don't think we have moved to C++17 yet. And if initialization is C++17 extension too, see: https://godbolt.org/z/Mc5TTMe8T. So I suggest changing this code to:
```
auto* r = p.value_if_ready();
if (r) {
...
} else {
...
}
```

Cc @ctiller

Reply:
probably just replace this with `p.value_if_ready() != nullptr`

Reply:
Done (first option), please take another look.

Reply:
Ah, sorry, I think @yashykt's suggestion is fair, i.e. change this to `if (p.value_if_ready() != nullptr) {`. At least we should not depend on implicit bool conversion.

Reply:
Done, please take another look

Reply:
This could just be spelled `if (p.ready()) {`

Reply:
Done, I now went with p.ready(), please take another look.
(Also, if you can trigger the remaining checks, that would be helpful, I'm unfamilar with how PRs are tested and merged in the grpc repo.)